echopype/convert/set_groups_base.py

```diff
@@ -210,3 +262,101 @@ def _beam_groups_vars(self):
         }
 
         return beam_groups_vars
+
+    @staticmethod
+    def _add_beam_dim(ds: xr.Dataset, sonar_model: str):
+        """
+        Adds ``beam`` as the last dimension to the appropriate
+        variables in ``Sonar/Beam_group1`` and ``Sonar/Beam_group2``
+        (when necessary).
+
+        Notes
+        -----
+        When expanding the dimension of a Dataarray, it is necessary
+        to copy the array (hence the .copy()). This allows the array
+        to be writable downstream (i.e. we can assign values to
+        certain indices).
+
+        To retain the attributes and encoding of ``beam``
+        it is necessary to use .assign_coords() with ``beam``
+        from ds.
+        """
+
+        # variables to add beam to
+        add_beam_names = set(ds.variables).intersection(
+            beam_only_names[sonar_model].union(beam_ping_time_names[sonar_model])
+        )
+
+        for var_name in add_beam_names:
+            if "beam" in ds.dims:
+                if "beam" not in ds[var_name].dims:
+                    ds[var_name] = (
+                        ds[var_name]
+                        .expand_dims(dim={"beam": ds.beam}, axis=ds[var_name].ndim)
+                        .assign_coords(beam=ds.beam)
+                        .copy()
+                    )
+            else:
+                # TODO: right now there is no attr or encoding for the beam dimension
+                #  if this changes in the future, we need to add them here.
+                ds[var_name] = (
```

Comment:
It may be worth adding a clarifying comment here, like this: "Add a single-value beam dimension"